plugins/uefi/fu-uefi-bootmgr.c

```diff
@@ -302,8 +302,11 @@ fu_uefi_bootmgr_bootnext (const gchar *esp_path,
 
 	/* if secure boot was turned on this might need to be installed separately */
 	source_app = fu_uefi_get_built_app_path (error);
-	if (source_app == NULL)
+	if (source_app == NULL) {
+		if (fu_efivar_secure_boot_enabled ())
+			g_prefix_error (error, "missing signed bootloader for secure boot: ");
 		return FALSE;
+	}
```

Comment:
...this is way too late. This is after the update has been deployed to the ESP -- checks that the update is going to be deployable need to be done in `->prepare_firmware()` not `->write_firmware()`. From a UX point of view the user is going to see it be "installed" and then for the process to fail. Will the update work? Should the user reboot? etc.

Reply:
Good point, thanks.  I just pushed a fixup.